### PR TITLE
output softmax set to fp32

### PR DESCRIPTION
**Pull request by ebgoldstein**
This PR fixes issue #11 - it sets all softmax in output layers to `dtype='float32'`.. should prevent numerical stability issues when using mixed precision training, which is default for Gym models

### CODE REVIEW

**Comment by ebgoldstein** [comment]
I am not sure if this NaN problem occurs with binary problems.. If it does, then i imagine the sigmoid activation is also a problem.. so that might need to be specifically cast as fp32 as well... (Note - that is not implemented here)

**Comment by ebgoldstein** [comment]
another relevant link (spaces added to avoid actual url link) : github.com/tensorflow/tensorflow/ issues/ 45568
